service/history/shardContext.go

```diff
@@ -56,6 +56,7 @@ type (
 		GetMetricsClient() metrics.Client
 		GetTimerAckLevel() time.Time
 		UpdateTimerAckLevel(ackLevel time.Time) error
+		Close()
```

Comment:
that's fine too. Today, shardController.historyShardItem only holds a ref to ExecutionFactory and not ExecutionManager. If we are doing this, then I will modify the historyShardsItem struct to hold executionManager instead of factory.